### DISCUSSION TITLE
revealjs presentation: hiding footer and slide menu control for a specific slide

### DISCUSSION THREAD

**Discussion by aborruso** [quarto-dev/quarto-cli#825] [Q&A]
Hi,
is it possible to hide the footer and slide menu control (the one you have a the bottom left), only for a slide?

It could be very useful especially for a slide that uses a full background image.

Thank you

**Comment by jjallaire** [comment]
This isn't possible right now --- if it was a simple CSS directive we could do it easily. Sadly the footer is in an entirely different layer so can't respond to CSS selectors on the main slide.

**Reply by cscheid** [reply]
@jjallaire could we install an event handler that checks for the presence-absence of a class and hides it when the current slide asks for it?

From https://revealjs.com/events/,

``` js
Reveal.on( 'slidechanged', event => {
  // event.previousSlide, event.currentSlide, event.indexh, event.indexv
} );
```

@aborruso in the meantime, I think this would also work for you, since you can just install that handler yourself too.

**Reply by jjallaire** [reply]
Yes, we definitely can do this sort of stuff w/ JS rather than CSS (e.g. see some of our existing code that manages footer content on slidechanged: https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/revealjs/plugins/support/support.js#L117). Just saying that we don't have time right now to implement this feature. It could definitely be done as a reveal plugin though (and we are hoping to make reveal plugins packagable extensions).

**Reply by cscheid** [reply]
Yup, that makes perfect sense. Here's what I thought. I haven't done extensive testing, but the following works for me:

```qmd
---
title: "Untitled"
format: 
  revealjs:
    slide-number: true
    include-after-body: reveal-events.html
---

## Slide 1

Hello.

## Slide 2 {.no-chrome}

Hide please.

## Slide 3

Show please.
```

And then `reveal-events.html` is:

```html
<script>
    Reveal.on('ready', () => {
        Reveal.on( 'slidechanged', event => {
            if (Array.from(event.currentSlide.classList).indexOf("no-chrome") !== -1) {
                document.querySelector("div .slide-number").style.visibility = "hidden";
            } else {
                document.querySelector("div .slide-number").style.visibility = null;
            }
        });
    })
</script>
```

@aborruso for your use case, you'll want to find the right part of the dom to hide (it'll be more than just `div .slide-number`)

**Reply by jjallaire** [reply]
Note that the the slide probably already has a class on it indicating it has a dark background (haven't checked to be sure though) so you might not even need the `.no-chrome` class

**Reply by aborruso** [reply]
> Yup, that makes perfect sense. Here's what I thought. I haven't done extensive testing, but the following works for me:

I have tested, but applying `.no-chrome` I have always the footer.

![image](https://user-images.githubusercontent.com/30607/167286288-a1cd4ffe-e6b3-43d1-b811-e53fec4a488b.png)


I have also tried to change the name of the class from `slide-number` to `footer`, but the footer still remains visible.

Thank you

```
<script>
  Reveal.on('ready', () => {
      Reveal.on( 'slidechanged', event => {
          if (Array.from(event.currentSlide.classList).indexOf("no-chrome") !== -1) {
              document.querySelector("div .footer").style.visibility = "hidden";
          } else {
              document.querySelector("div .footer").style.visibility = null;
          }
      });
  })
</script>
```

**Reply by jjallaire** [reply]
I don't think that was intended as a literal "paste in this code and you'll get the behavior you want". @cscheid was just illustrating the general idea of registering a Reveal event handler -- In fact if you just pasted that into a document it wouldn't even run (Reveal not defined JS error) b/c Reveal isn't loaded until after the document body. 

The way to really do something like this is with a Reveal plugin, which definitely requires some knowledge of JavaScript and CSS as well as the Reveal DOM structure and event system. So the net is that this is possible if you know JS/CSS and are willing to put some time in hacking. Failing that we will definitely come back to this later on this year when we do another round of improvements to Reveal.

**Reply by aborruso** [reply]
> I don't think that was intended as a literal "paste in this code and you'll get the behavior you want"

@jjallaire  I'm sorry, I didn't understand, thanks.

I thought it was code I could use, because he writes "the following works for me". 

Sorry again

**Reply by jjallaire** [reply]
Actually I am wrong and this is indeed intended as runnable (I missed the part about it being included via `include-after-body`, that. makes the JS run without errors). But he also notes that this only hides the slide number and there may be other DOM elements you need to go after -- I see that you've attempted to do that and it didn't work. We would (honestly!) love to keep hacking on this and figure it out but are just too tied up with other stuff right now. If you experiment (and esp. if you have some JS/CSS experience) you may be able to get this working. If not, fear not, we will come back to this later on this year and make it easy.

**Comment by cderv** [comment]
I have field an issue about this feature request to track https://github.com/quarto-dev/quarto-cli/issues/1445 as it also came up in https://github.com/quarto-dev/quarto-cli/issues/1391#issuecomment-1186148935

**Comment by mcanouil** [comment]
FYI:
- Opened issue:
  - #1445
- Related discussions:
  - #557
  - #825
  - #4181
  - #5566

**Reply by aborruso** [reply]
Thank you very much @mcanouil

**Comment by EllaKaye** [comment]
As of Quarto v1.6, you can hide the footer on the title slide with this in the top yaml:

```
title-slide-attributes:
  data-footer: "false"
```

**Reply by aborruso** [reply]
Thank you very much
